cmd/snap/cmd_run.go

```diff
@@ -1036,7 +1058,23 @@ func (x *cmdRun) runSnapConfine(info *snap.Info, securityTag, snapApp, hook stri
 
 	logger.Debugf("executing snap-confine from %s", snapConfine)
 
-	snapName, _ := snap.SplitSnapApp(snapApp)
+	snapName, appName := snap.SplitSnapApp(snapApp)
+	if hook == "" {
+		clean, err := maybeWaitWhileInhibited(snapName)
+		if err != nil {
+			return err
+		}
+		// make sure to release inhibition file lock and finish RAA notification
+		// flow before returning.
+		defer clean()
```

Comment:
I don't think we can use defer to finish the flow, as at the end of this chain of functions we do exec which will not return, no?